# SPDX-License-Identifier: BSD-2-Clause
# Origin: EmberBSD (AI-assisted), explicit language for libiconv's C rules.
# Shared pkgsrc Libtool may have a different configured compiler spelling
# (for example a language-standard flag). Keep the consumer's compiler and
# select its C tag explicitly, as Automake-generated rules already do.
--- a/converters/libiconv/Makefile
+++ b/converters/libiconv/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.72 2026/07/23 14:56:17 ryoon Exp $
 
 DISTNAME=	libiconv-1.19
+PKGREVISION=	1
 CATEGORIES=	converters
 MASTER_SITES=	${MASTER_SITE_GNU:=libiconv/}
 
@@ -15,6 +16,14 @@
 GNU_CONFIGURE=		yes
 GNU_CONFIGURE_STRICT=	no	# has multiple configure scripts
 
+SUBST_CLASSES+=		libtool-tag
+SUBST_STAGE.libtool-tag=	pre-configure
+SUBST_MESSAGE.libtool-tag= Select the C Libtool tag explicitly.
+SUBST_FILES.libtool-tag=	lib/Makefile.in libcharset/lib/Makefile.in \
+			src/Makefile.in tests/Makefile.in
+SUBST_SED.libtool-tag=	-e 's|$$(LIBTOOL) --mode=compile|$$(LIBTOOL) --tag=CC --mode=compile|g' \
+			-e 's|$$(LIBTOOL) --mode=link|$$(LIBTOOL) --tag=CC --mode=link|g'
+
 CONFLICTS=	man-pages-[0-9]*
 
 MAKE_JOBS_SAFE=	no
